public/pages/Rules/utils/helpers.tsx

```diff
@@ -177,3 +177,19 @@ export function setBreadCrumb(
 ) {
   breadCrumbSetter?.([BREADCRUMBS.SECURITY_ANALYTICS, BREADCRUMBS.RULES, breadCrumb]);
 }
+
+export function getLogTypeFromLogSource(logSource: { [k: string]: string }) {
+  const logTypes = new Set(ruleTypes.map(({ value }) => value));
+
+  let logType;
+  ['product', 'category', 'service'].some((key) => {
+    if (logSource[key] && logTypes.has(logSource[key])) {
+      logType = logSource[key];
+      return true;
```

Comment:
Minor: we can then return logSource[key] directly, and null if no matching logType is found

Reply:
Simplified the code